zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/ProcessInstanceControllerTest.java

```diff
@@ -406,4 +410,117 @@ void shouldRejectCreateProcessInstancesIfOperationReferenceIsNotValid() {
         .expectBody()
         .json(expectedBody);
   }
+
+  @Test
+  void shouldCancelProcessInstance() {
+    // given
+    when(processInstanceServices.cancelProcessInstance(any(ProcessInstanceCancelRequest.class)))
+        .thenReturn(CompletableFuture.completedFuture(new ProcessInstanceRecord()));
+
+    final var request =
+        """
+        {
+          "operationReference": 123
+        }""";
+
+    // when/then
+    webClient
+        .post()
+        .uri(CANCEL_PROCESS_URL.formatted("1"))
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isNoContent();
+
+    Mockito.verify(processInstanceServices).cancelProcessInstance(cancelRequestCaptor.capture());
+    final var capturedRequest = cancelRequestCaptor.getValue();
+    assertThat(capturedRequest.processInstanceKey()).isEqualTo(1);
+    assertThat(capturedRequest.operationReference()).isEqualTo(123L);
+  }
+
+  @Test
+  void shouldCancelProcessInstanceWithNoBody() {
+    // given
+    when(processInstanceServices.cancelProcessInstance(any(ProcessInstanceCancelRequest.class)))
+        .thenReturn(CompletableFuture.completedFuture(new ProcessInstanceRecord()));
+
+    // when/then
+    webClient
+        .post()
+        .uri(CANCEL_PROCESS_URL.formatted("1"))
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus()
+        .isNoContent();
+
+    Mockito.verify(processInstanceServices).cancelProcessInstance(cancelRequestCaptor.capture());
+    final var capturedRequest = cancelRequestCaptor.getValue();
+    assertThat(capturedRequest.processInstanceKey()).isEqualTo(1);
+    assertThat(capturedRequest.operationReference()).isNull();
+  }
+
+  @Test
+  void shouldDeleteResourceWithEmptyBody() {
+    // given
+    when(processInstanceServices.cancelProcessInstance(any(ProcessInstanceCancelRequest.class)))
+        .thenReturn(CompletableFuture.completedFuture(new ProcessInstanceRecord()));
+
+    final var request =
+        """
+        {}""";
+
+    // when/then
+    webClient
+        .post()
+        .uri(CANCEL_PROCESS_URL.formatted("1"))
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isNoContent();
+
+    Mockito.verify(processInstanceServices).cancelProcessInstance(cancelRequestCaptor.capture());
+    final var capturedRequest = cancelRequestCaptor.getValue();
+    assertThat(capturedRequest.processInstanceKey()).isEqualTo(1);
+    assertThat(capturedRequest.operationReference()).isEqualTo(null);
```

Comment:
🙃 Just a small change
```suggestion
    assertThat(capturedRequest.operationReference()).isNull();
```